### PR TITLE
Include inferred `joinable!` in `diesel print-schema`

### PR DESCRIPTION
**Pull request by sgrif**
This was a pretty straightforward addition. I've not included an
explicit test that the filtering is applied, or that the unsafe
constraints are removed here. It may be worth more exhaustively testing
this.

### CODE REVIEW

File: diesel_cli/src/print_schema.rs [hunk 1]
```diff
@@ -67,6 +75,14 @@ impl Display for TableDefinitions {
             }
             writeln!(f, "{}", TableDefinition(table))?;
         }
+
+        if !self.1.is_empty() {
+            write!(f, "\n")?;
+        }
+
+        for foreign_key in &self.1 {
+            write!(f, "{}", Joinable(foreign_key))?;
```
Comment (killercup) [inline diesel_cli/src/print_schema.rs]: I'd use `writeln!` here and `write!` in the `<Joinable<'a> as Display>::fmt` method

File: diesel_cli/tests/print_schema.rs [hunk 2]
```diff
@@ -390,3 +390,80 @@ r"pub mod custom_schema {
...
 ");
 }
+
+#[test]
+#[cfg(feature = "postgres")]
+fn print_schema_with_foreign_keys() {
+    let p = project("print_schema_with_foreign_keys").build();
+    let db = database(&p.database_url());
+
+    // Make sure the project is setup
```
Comment (killercup) [inline diesel_cli/tests/print_schema.rs]: Can't help but notice the redundant comment that is also spelt wrong (should be "is set up") which was successfully cargo culted into every single test case 😆

File: diesel_cli/src/print_schema.rs [hunk 3]
```diff
@@ -118,6 +134,20 @@ impl<'a> Display for ColumnDefinitions<'a> {
     }
 }
 
+struct Joinable<'a>(&'a ForeignKeyConstraint);
+
+impl<'a> Display for Joinable<'a> {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        writeln!(
+            f,
+            "joinable!({} -> {} ({}));",
```
Comment (killercup) [inline diesel_cli/src/print_schema.rs]: I see you're using rustfmt style now. I'd put the first two args on the first line, but whatever

Comment (sgrif) [inline diesel_cli/src/print_schema.rs]: I'm going to run it on the whole codebase one of these days. Trying to reduce the churn in the short term

File: diesel_cli/tests/print_schema.rs (hunk 2 above)
Comment (sgrif) [inline diesel_cli/tests/print_schema.rs]: Lol I'll get rid of it

**Pull request merged**